### PR TITLE
feat: materialize date-specific plans from routines

### LINKED ISSUE #19: [Feature] Materialize date-specific Plans from Routines and support one-day overrides
Parent Epic: #17
Depends on: #18

## Issue type
Feature

## Goal
Introduce/complete the date-specific Plan lifecycle so the user can change today's intention without rewriting the recurring Routine.

## Required behavior
For a given date, FrilDay must be able to resolve the effective Plans from active Routine schedule rules and persisted date-specific changes.

A Plan must preserve the intended planned duration for that date even if the Routine default changes later, according to the chosen materialization strategy.

## One-day overrides
Support at minimum:

- change planned duration for a specific date
- skip a specific date
- restore/remove the override to return to Routine-derived behavior where meaningful

A one-day override must not mutate the Routine's recurring defaults.

## Materialization strategy
Choose and document one deterministic strategy, for example:

- eagerly persist generated Plans for a planning horizon, or
- resolve virtual Routine-derived Plans and persist only explicit overrides/execution references.

Either strategy is acceptable if it guarantees stable history and avoids duplicate Plans.

The implementation must answer:

- what identifies a Plan uniquely?
- when is a Routine-derived Plan considered created?
- how does a Session reference a Plan that was initially virtual?
- what happens if a Routine schedule changes after a historical Plan was executed?
- how are skipped/moved Plans represented?

## Historical stability
Once a Plan has execution history (Session/Completion), later Routine changes must not make that historical execution disappear or point at a different intended duration silently.

## Migration
Map current schedule-derived Today/weekly behavior to the new Plan model without destructive loss of existing completions/time entries.

## Acceptance criteria
- [ ] A scheduled Routine resolves to an executable date-specific Plan.
- [ ] Specific-day duration override does not change the Routine default.
- [ ] A date can be skipped explicitly.
- [ ] Plan identity/materialization strategy is documented and tested.
- [ ] No duplicate Plans are produced across reloads/resolution.
- [ ] Routine edits do not erase historical executed Plans.
- [ ] Sessions/Completions can reference stable historical intent.
- [ ] Existing history has a migration/mapping path.
- [ ] Rust tests cover schedule/override/history edge cases.

## Out of scope
- visual weekly planner polish
- drag/drop calendar
- cloud conflict resolution

### PR DESCRIPTION
**Pull request by mors119**
## Summary
- resolve virtual Routine-derived Plans with deterministic `routine-plan:<utf8-length>:<routine-id>:<date>` identity
- persist one-day duration overrides, skips, and execution/completion snapshots without mutating Routine defaults
- connect Session and Completion records to Plan identity and use effective duration for planned-vs-actual timer feedback
- migrate existing SQLite and legacy localStorage history by backfilling stable Plan records
- expose Today controls and reflect effective Plans in weekly schedule

## Verification
- `cargo test --workspace`
- `bun run build` (from `apps/desktop`)
- `bun test` (from `apps/desktop`)
- `bun run lint` (from `apps/desktop`)

Closes #19

### CODE REVIEW

File: apps/desktop/src/app/store/useFrilDayStore.ts [hunk 1]
```diff
@@ -204,6 +236,94 @@ export const useFrilDayStore = create<FrilDayState>((set, get) => ({
...
+    if (hasPlanHistory(get(), id, taskId, date)) {
+      set({ errorMsg: 'Historical plans cannot be changed.' });
+      return false;
+    }
+    const nextPlan = createRoutinePlan({
+      routineId: taskId,
+      date,
+      baselineDurationMinutes: current?.baselineDurationMinutes ?? task.durationMinutes,
+      durationOverrideMinutes: current?.durationOverrideMinutes,
+      status: 'skipped',
```
[the remark this answers is not in the document]
Comment (mors119) [reply apps/desktop/src/app/store/useFrilDayStore.ts:300]: Addressed in 37dfbc3. core_statistics now receives persisted Plans, and completion statistics resolve effective executable Plans so skipped occurrences are excluded from daily, weekly, and period denominators. Added Rust regression coverage and desktop adapter coverage.
